### PR TITLE
Stand-alone manifest for CVA6

### PR DESCRIPTION
**Pull request by MikeOpenHWGroup**
Please do not merge this (yet).

The primary purpose of this pull-request is to review `cva6_manifest.flist`, a stand-alone manifest for the CVA6 core (without any of the external TB or FPGA components).  This manifest is currently used in core-v-verif (see [core-v-verif #451](https://github.com/openhwgroup/core-v-verif/pull/451)), so I am at least 90% confident that is it correct and complete.

Ultimately this manifest should be in a better form such as a YAML structure, but let's focus on content first.  I have three specific comments/questions:
1. Is the list of files in the manifest complete?
2. The RTL should not be dependent on a file in the tb directory.  So `tb/ariane_soc_pkg.sv` will need to find another home, or the RTL-specific content could be extracted and moved to a file in another directory.
3. The core RTL should not be dependent on an FPGA-specific module.  This affects `src/fpga-support/rtl/SyncSpRamBeNx64.sv`.

The other changes in this pull-request exist to support Metrics dsim.  I am still working through some of this and will open specific issues and/or pull-requests on this later.

### CODE REVIEW

**Comment by LucaZulberti** [comment]
Hi @MikeOpenHWGroup, is the solution given by #598 a good starting point for your second comment?

File: tb/ariane_tb.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #include "verilator.h"
 #include "verilated.h"
 #include "verilated_vcd_c.h"
-#include "Variane_testharness__Dpi.h"
+//#include "Variane_testharness__Dpi.h"
```
Comment (zarubaf) [inline tb/ariane_tb.cpp:22]: Why is this needed?

**Review by zarubaf — COMMENTED** [review]
> Ultimately this manifest should be in a better form such as a YAML structure, but let's focus on content first. 
Yes, please - maybe we can also use something standardized - this is effectively manifest number 5 in this repo...
> I have three specific comments/questions:
> 
> 1. Is the list of files in the manifest complete?
Looks okay for the write-through cache version. Of course, the write-back cache needs different files.
> 2. The RTL should not be dependent on a file in the tb directory.  So `tb/ariane_soc_pkg.sv` will need to find another home, or the RTL-specific content could be extracted and moved to a file in another directory.
See #598.
> 3. The core RTL should not be dependent on an FPGA-specific module.  This affects `src/fpga-support/rtl/SyncSpRamBeNx64.sv`.
It is a behavioral description of an SRAM, just a bit tuned so that the Xilinx and Altera tools can infer BRAMs for it. The `fpga-support` repo is anyway deprecated afaik, so we can flatten and re-name if that helps the situation. Don't think this is really high-priority, tbh.

File: tb/ariane_tb.cpp (hunk 1 above)
Comment (MikeOpenHWGroup) [reply tb/ariane_tb.cpp]: Opps.  Good catch, thanks.  That should not have been pushed in.  Restored.

**Comment by MikeOpenHWGroup** [comment]
> Ultimately this manifest should be in a better form such as a YAML structure, but let's focus on content first.
>> Yes, please - maybe we can also use something standardized - this is effectively manifest number 5 in this repo...

Completely agree.  Standardizing the manifests is a priority for me.  I feel bad for adding manifest number 5 and promise to eliminate it as soon as possible.

> 3. The core RTL should not be dependent on an FPGA-specific module. This affects src/fpga-support/rtl/SyncSpRamBeNx64.sv.
>> It is a behavioral description of an SRAM, just a bit tuned so that the Xilinx and Altera tools can infer BRAMs for it. The fpga-support repo is anyway deprecated afaik, so we can flatten and re-name if that helps the situation. Don't think this is really high-priority, tbh.

No time like the present.  :wink: Either we fix it now, or I will open another specific issue for it.
